### PR TITLE
Populate the publisher with required course data from course_metadata

### PR DESCRIPTION
**Pull request by schenedx**
Update the publisher data loaders so all required data on the course is imported into publisher from course discovery. This include course images
EDUCATOR-676

### CODE REVIEW

**Comment by schenedx** [comment]
@clintonb Please help review my change to update the management command to populate data in Publisher again

File: course_discovery/apps/publisher/dataloader/create_courses.py [hunk 1]
```diff
@@ -33,15 +37,14 @@ def process_course(meta_data_course):
         create_or_update_course(meta_data_course, available_organization)
 
     except:  # pylint: disable=bare-except
-        logger.error('Exception appear for course-id [%s].', meta_data_course.uuid)
+        logger.error('Exception appear for course-id [%s]. %s', meta_data_course.uuid, traceback.format_exc())
```
Comment (clintonb) [inline course_discovery/apps/publisher/dataloader/create_courses.py]: Use `logger.exception()` and ditch the usage of `traceback`.

```python
logger.exception('An error occurred while processing course [%s]!', meta_data_course.uuid)
```

File: course_discovery/apps/publisher/dataloader/create_courses.py [hunk 2]
```diff
@@ -23,7 +28,6 @@ def process_course(meta_data_course):
     # if course has more than 1 organization don't import that course. Just log the entry.
     # that course will import manually.
     organizations = meta_data_course.authoring_organizations.all()
-
```
Comment (clintonb) [inline course_discovery/apps/publisher/dataloader/create_courses.py]: Restore this new line to keep things legible.

File: course_discovery/apps/publisher/dataloader/create_courses.py [hunk 3]
```diff
@@ -33,15 +37,14 @@ def process_course(meta_data_course):
...
-        logger.error('Exception appear for course-id [%s].', meta_data_course.uuid)
+        logger.error('Exception appear for course-id [%s]. %s', meta_data_course.uuid, traceback.format_exc())
 
 
 def create_or_update_course(meta_data_course, available_organization):
 
     primary_subject = None
     secondary_subject = None
     tertiary_subject = None
-
```
Comment (clintonb) [inline course_discovery/apps/publisher/dataloader/create_courses.py]: Restore the newline.

File: course_discovery/apps/publisher/dataloader/create_courses.py [hunk 4]
```diff
@@ -57,14 +60,19 @@ def create_or_update_course(meta_data_course, available_organization):
         'level_type': meta_data_course.level_type,
         'primary_subject': primary_subject, 'secondary_subject': secondary_subject,
         'tertiary_subject': tertiary_subject,
-        'video_link': meta_data_course.video.src if meta_data_course.video else None
+        'video_link': meta_data_course.video.src if meta_data_course.video else None,
+        'expected_learnings': meta_data_course.outcome,
+        'prerequisites': meta_data_course.prerequisites_raw,
+        'syllabus': meta_data_course.syllabus_raw
```
Comment (clintonb) [inline course_discovery/apps/publisher/dataloader/create_courses.py]: Add a trailing comma.

File: course_discovery/apps/publisher/dataloader/create_courses.py [hunk 5]
```diff
@@ -84,6 +92,26 @@ def create_or_update_course(meta_data_course, available_organization):
...
+        )
+    elif meta_data_course.card_image_url:
+        response = requests.get(meta_data_course.card_image_url)
+        if response.status_code == 200:
+            img_name = meta_data_course.card_image_url.split('/')[-1]
+            with io.BytesIO() as fp:
+                fp.write(response.content)
+                publisher_course.image.save(img_name, File(fp))
+        else:
+            logger.warning('Image download from {} failed because status [{}]'.format(
```
Comment (clintonb) [inline course_discovery/apps/publisher/dataloader/create_courses.py]:
```python
logger.error('Failed to download image for course [%s] from [%s]. Server responded with status [%d].', meta_data_course.uuid, meta_data_course.card_image_url, response.status_code)```

**Review by clintonb — APPROVED** [review]
Please address feedback before merging.

**Comment by schenedx** [comment]
@clintonb Feedbacks addressed. Thank you!

**Pull request merged**